### PR TITLE
Jenish/feature/iris get logs

### PR DESCRIPTION
**Pull request by JenishPatel99**

### CODE REVIEW

File: ex2_hal/iris/hardware_interface/include/iris.h [hunk 1]
```diff
@@ -22,11 +22,12 @@
...
 // Iris data transfer constants
-#define IMAGE_TRANSFER_SIZE 512 // 512 data bytes
-#define MAX_IMAGE_LENGTH 3      // In bytes
-#define MAX_IMAGE_COUNT 1       // In bytes
-#define HOUSEKEEPING_SIZE 23    // In bytes
-#define IRIS_UNIX_TIME_SIZE 4   // In bytes
+#define IMAGE_TRANSFER_SIZE 512    // 512 data bytes
+#define IRIS_LOG_TRANSFER_SIZE 512 // 512 data bytes
+#define MAX_IMAGE_LENGTH 3         // In bytes
+#define MAX_IMAGE_COUNT 1          // In bytes
```
Comment (Taylorr82) [inline ex2_hal/iris/hardware_interface/include/iris.h:28]: In bytes?

File: ex2_hal/iris/hardware_interface/include/iris.h [hunk 2]
```diff
@@ -41,7 +42,8 @@
 #define IRIS_POWER_CYCLE_DELAY vTaskDelay(pdMS_TO_TICKS(1))
 #define IRIS_INIT_DELAY vTaskDelay(pdMS_TO_TICKS(4000))
 #define IRIS_WAIT_FOR_STATE_TRANSITION vTaskDelay(pdMS_TO_TICKS(100))
-#define IRIS_IMAGE_DATA_BLOCK_TRANSFER_DELAY vTaskDelay(pdMS_TO_TICKS(10)) // Depends on optimization level
+#define IRIS_IMAGE_DATA_BLOCK_TRANSFER_DELAY vTaskDelay(pdMS_TO_TICKS(20)) // Depends on optimization level
```
Comment (Taylorr82) [inline ex2_hal/iris/hardware_interface/include/iris.h:45]: Depends on who's optimization level

**Review by Taylorr82 — APPROVED** [review]

File: ex2_hal/iris/hardware_interface/include/iris.h (hunk 1 above)
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/include/iris.h]: Sorry that was a typo

File: ex2_hal/iris/hardware_interface/include/iris.h (hunk 2 above)
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/include/iris.h]: The optimization level of the OBC's compiler, i have removed the comment and will test out the data transfer at o3 to see if any modifications is necessary. Basically whatever delay that is chosen will work with the o3 level

**Review by Taylorr82 — APPROVED** [review]

File: ex2_services/Services/source/payload/iris/iris_service.c [hunk 3]
```diff
@@ -176,13 +177,21 @@ SAT_returnState iris_service_app(csp_packet_t *packet) {
...
+        uint16_t image_count = 0;
+        char filename[REDCONF_NAME_MAX];
 
-        status = iris_get_image_length(&image_length);
-        IRIS_SERVICE_IMAGE_TRANSFER_DELAY; // 100 ms delay
+        status = iris_get_image_count(&image_count);
+        if (status == IRIS_HAL_OK) {
+            for (int i = 0; i < image_count; i++) {
+                status = iris_get_image_length(&image_length);
+                IRIS_SERVICE_IMAGE_TRANSFER_DELAY;
```
Comment (Taylorr82) [inline ex2_services/Services/source/payload/iris/iris_service.c:188]: What is this delay for?

**Review by Taylorr82 — DISMISSED** [review]

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**